arm/Microsoft.Synapse/workspaces/.bicep/nested_cmkRbac.bicep

```diff
@@ -0,0 +1,39 @@
+param keyvaultName string
+param workspaceIdentity string
+param usesRbacAuthorization bool = false
+
+// Workspace encryption - Assign Workspace System Identity Keyvault Crypto Reader at Encryption Keyvault
+resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
+  name: keyvaultName
```

Comment:
This is a general comment for us to figure out how to deal with key vault dependency, which must be a purge protected key vault

Reply:
What shall I do here?

Reply:
@eriqua any thoughts? I guess we won't get around enabling purge protection

Reply:
I guess so. I'm wondering how to deal with that for the Spark lab. Attendees will run the dependency pipeline and stay with 3 kv they won't be able to purge. Luckily kv does not have costs but still maybe not ideal.

- Question: what if we reduced the `softDeleteRetentionInDays` to the minimum? I understand from the docs that the minimum is 7. Would that be a workaround to at least have the deleted kv removed shortly?
- Alternatively, or additionally, we could add a boolean flag to disable the deployment of synapse dependencies by default as done for the sqlmi and image dependencies

Reply:
Unless synapse need a dedicated kv I agree. The kv with nopurge is being worked on as part of PR #773. I'll let you know once ready

Reply:
We now have the purge-protected key vault: `adp-carml-az-kv-nopr-002`